src/SonarScanner.MSBuild.Shim/SonarEngineWrapper.cs

```diff
@@ -0,0 +1,62 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2025 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarScanner.MSBuild.Shim;
+
+public class SonarEngineWrapper
+{
+    private readonly IRuntime runtime;
+    private readonly IProcessRunner processRunner;
+
+    public SonarEngineWrapper(IRuntime runtime, IProcessRunner processRunner)
+    {
+        this.runtime = runtime ?? throw new ArgumentNullException(nameof(runtime));
+        this.processRunner = processRunner ?? throw new ArgumentNullException(nameof(processRunner));
+    }
+
+    public virtual bool Execute(AnalysisConfig config, string propertiesJsonInput)
+    {
+        _ = config ?? throw new ArgumentNullException(nameof(config));
+
+        return InternalExecute(config, propertiesJsonInput);
+    }
+
+    private bool InternalExecute(AnalysisConfig config, string propertiesJsonInput)
```

Comment:
I called it like this on other places
```suggestion
    private bool InternalExecute(AnalysisConfig config, string engineInput)
```
or like this, but I find that redundant in this class
```suggestion
    private bool InternalExecute(AnalysisConfig config, string scannerEngineInput)
```
In this class, it could also be just `input` or `standardInput` - the same, as in the `ProcessRunnerArguments`